File: 프로그래머스/Level 1/옹알이2.swift
import Foundation

func solution(_ babbling:[String]) -> Int {
    var answer:Int = 0
    var words:[String] = ["aya", "ye", "woo", "ma"]
    
    for babb in babbling {
        var b = babb
        for word in words {
            let tmp = word + word
            if !b.contains(tmp) {
                b = b.replacingOccurrences(of: word, with: " ")
            }
        }
        print(b)
        if b.isAllSpace() {
            answer += 1
        }
    }
    return answer
}

extension String {
    func isAllSpace() -> Bool {
        return allSatisfy { $0.isWhitespace}
    }
}

File: 프로그래머스/Level 1/옹알이2.py
# https://school.programmers.co.kr/learn/courses/30/lessons/133499
def solution(babbling):
    answer = 0
    words = ["aya", "ye", "woo", "ma"]

    for babb in babbling:
        for word in words:
            if word*2 not in babb:
                babb = babb.replace(word, " ")

        if babb.isspace():
            answer += 1
    return answer
